### ISSUE TITLE
Use $_SERVER['argv'] when $argv is not available.

### ISSUE THREAD

**Issue by Beanow** [friendica/friendica#1218]
`$argv` and `$argc` require `register_argc_argv=On` in your php.ini file.
But `$_SERVER['argv']` does not.

In the **global scope** you could do this as a quick fix:
```php
// Populate $argv and $argc.
if(!isset($argv) && array_key_exists('argv', $_SERVER)){
  $argv = $_SERVER['argv'];
  $argc = count($argv);
}
```

But better would be to remove the dependency for `$argv` and `$argc` from the global scope.
This may require more code refactoring though.

**Update:** I noticed there is a `$a->argv`, but this is not the same as the PHP `$argv`!!
We cannot use this to store $argv.

**Comment by annando** [comment]
I created a pull request for that. Question is, if there is any case where $argc and $argv are available and the corresponding $_SERVER variables aren't.

**Comment by Beanow** [comment]
When `register_argc_argv` is set to `Off` only the `$_SERVER` values are available I believe.

**Comment by annando** [comment]
I think so too. But I haven't found a documentation that told me that there *never* would be situation where *only* $argc and $argv are existing.

**Comment by Beanow** [comment]
The `!isset($argv)` will safeguard this scenario. The only time where this code would not work is when both `$argv` and `$_SERVER['argv']` do not exist. You can add a more clear error message in that case.

**Comment by fabrixxm** [comment]
Closed by #1274
